rocketpy/motors/HybridMotor.py

```diff
@@ -493,6 +493,7 @@ def addTank(self, tank, position):
         """
         self.liquid.addTank(tank, position)
         self.solid.massFlowRate = self.totalMassFlowRate - self.liquid.massFlowRate
+        reset_funcified_methods(self)
```

Comment:
So every time a Tank is added to a motor every cached_property of that motor object gets reset?

This probably works for the Motor classes but it can be dangerous in other places

Reply:
Could you elaborate more what do you mean by "dagerous"? My first though was exactly the opposite, all the cached parameters are being reset, even if some don't need it, therefore the code is playing safe here to ensure correctness.

Reply:
We might reset a property that does not need recalculation and is computationally heavy to calculate. Eventually, that property could need to be recalculated just because of the unnecessary reset. 

I can't think of an example where this would happen though

Reply:
I agree with you, although this is not likely to happen in methods that change fundamentally the instance, such as addTank, since most of the cache would need to be updated anyway.

I don't have a good solution for the scenario that you described. In the short term, the `del` can be used to select specific attributes to be deleted.

Reply:
I think it is not dangerous, instead, it is inefficient.  

We don't need to be fast now, let's wait to see if any part of the code starts to increase its runtime due to this function.
We can also address it in the future.

Reply:
Not everytime.... Only when the methods have already been cached. Methods are only cached once called. In the general RocketPy workflow, you define everything before actually calling funcified methods.